### ISSUE TITLE
`manual_unwrap_or` is not showing for simple ref cases

### ISSUE THREAD

**Issue by nyurik** [rust-lang/rust-clippy#11795]
### Summary

This case is not detected. I suspect this is due to the initial fix for #7228 (?).

### Lint Name

manual_unwrap_or

### Reproducer

I tried this code:

```rust
fn test_func(val: Option<&str>) -> String {
    let s = match val {
        None => "",
        Some(v) => v,
    };
    s.to_string()
}
```

I expected to see this happen: a warning to simplify the match statement to use `unwrap_or`

Instead, this happened: no warnings

### Version

```text
rustc 1.73.0 (cc66ad468 2023-10-03)
binary: rustc
commit-hash: cc66ad468955717ab92600c770da8c1601a4ff33
commit-date: 2023-10-03
host: x86_64-unknown-linux-gnu
release: 1.73.0
LLVM version: 17.0.2
```

**Comment by nyurik** [comment]
P.S. Note that the above could also be solved with `unwrap_or_default()`
